### ISSUE TITLE
Make the Command Line Mode easier to access

### ISSUE THREAD

**Issue by Don-Vito** [microsoft/terminal#8322]
# Description of the new feature/enhancement
Today, to get to the command line mode I need to open the action mode and then to delete a ">" symbol, which is not very convenient.

While discussing this with @zadjii-msft few ideas were raised:
1. Adding a separate binding for ToggleCommandPalette(mode=commandline)
2. Having ToggleCommandPalette() switch between modes: closed->action->command line->closed.
3. Both of them ☺️ 

# Proposed technical implementation details (optional)
The most straightforward solution is to add an optional command palette mode to the ToggleCommandPalette, with two "launch modes": action and command line.

We will be able to implement option 2 only once https://github.com/microsoft/terminal/pull/8044 gets merged.

**Comment by zadjii-msft** [comment]
I'll be danged, I could have sworn I had already filed a task for this.

Ideas:
* I don't really care that toggle with a mode might also close it. Maybe if you're in action mode, and you press the `toggleCommandPalette(mode=commandline)` keybinding, it'll pop you directly into commandline mode? (with #8044) (and vice versa)
* Action and commandline mode definitely, tab search maybe? I suppose `{ action: toggleCommandPalette, mode: tabSearch }` doesn't _really_ make sense now does it.

**Comment by DHowett** [comment]
I'd say that the command palette being the backend for tab search is an implementation detail, not something to expose to the users :smile:

**Comment by Don-Vito** [comment]
> 
> 
> I'll be danged, I could have sworn I had already filed a task for this.
> 
> Ideas:
> 
>     * I don't really care that toggle with a mode might also close it. Maybe if you're in action mode, and you press the `toggleCommandPalette(mode=commandline)` keybinding, it'll pop you directly into commandline mode? (with #8044) (and vice versa)
> 
>     * Action and commandline mode definitely, tab search maybe? I suppose `{ action: toggleCommandPalette, mode: tabSearch }` doesn't _really_ make sense now does it.

@zadjii-msft  - I believe that:
- toggleCommandPalette(mode=commandline) - I wish I could make it work from everywhere (even if the window is minimized and the PC is hibernated)
- interesting alternative is to use the toggleCommandPalette binding to switch between the modes. I mean, the first ctrl+shift+p opens the action mode, the second one switches to commandline mode

**Comment by Don-Vito** [comment]
> 
> 
> I'd say that the command palette being the backend for tab search is an implementation detail, not something to expose to the users 😄

Yep.. that was a bad idea, especially since we already have a tabSeach binding...

**Comment by Don-Vito** [comment]
> 
> I'll be danged, I could have sworn I had already filed a task for this.
> 

@zadjii-msft - I found it.. in the form of TODO 😆 

```
 void TerminalPage::_HandleToggleCommandPalette(const IInspectable& /*sender*/,
                                                   const ActionEventArgs& args)
    {
        // TODO GH#6677: When we add support for commandline mode, first set the
        // mode that the command palette should be in, before making it visible.
        CommandPalette().EnableCommandPaletteMode();
        CommandPalette().Visibility(CommandPalette().Visibility() == Visibility::Visible ?
                                        Visibility::Collapsed :
                                        Visibility::Visible);
        args.Handled(true);
    }
```

**Comment by Jspooner09** [comment]
I have computer virus


Jimmy R. 

CONFIDENTIALITY: The information contained in this transmission may contain privileged and confidential information. It is intended only for the use of the person(s) named above. If you are not the intended recipient, you are hereby notified that any review, dissemination, distribution or duplication of this communication, and the information contained in it, is strictly prohibited. If you are not the intended recipient, please contact the sender and immediately destroy all copies of the original message.



On Thursday, December 3, 2020, 11:16 AM, msftbot[bot] <notifications@github.com> wrote:




Closed #8322 via f072eaf.
